include/networkit/components/ConnectedComponentsImpl.hpp

```diff
@@ -0,0 +1,37 @@
+// networkit-format
+#ifndef NETWORKIT_COMPONENTS_CONNECTED_COMPONENTS_IMPL_HPP_
+#define NETWORKIT_COMPONENTS_CONNECTED_COMPONENTS_IMPL_HPP_
+
+#include <networkit/Globals.hpp>
+#include <networkit/graph/Graph.hpp>
+#include <networkit/structures/Partition.hpp>
+
+namespace NetworKit {
+namespace ConnectedComponentsDetails {
+
+class ConnectedComponentsImpl {
```

Comment:
Why doesn't this class inherit from `Algorithm` but instead re-implements parts of its interface (`hasRun`, `run`, `assureFinished`)?